monsteer/plugin/spikeReport.cpp

```diff
@@ -49,19 +50,26 @@ namespace
 const uint32_t INTERNAL_TIMEOUT = 500;
 }
 
+URI toHostAndPort( const URI& uri )
+{
+    URI out;
+    out.setHost( uri.getHost( ));
+    out.setPort( uri.getPort( ));
+    return out;
+}
+
 SpikeReport::SpikeReport( const brion::SpikeReportInitData& pluginData )
-    : _uri( pluginData.getURI( ))
+    : _uri( toHostAndPort( pluginData.getURI( )))
```

Comment:
I think it's better not to use a function here. You still need to default construct an URI inside your function to get the job done so you're not saving any work using the default initializer. Even worse, the name actually confused me. Since what you're doing is simply copying over the host and port I think you can simply put that in the constructor body, the intent is clear and it's much more concise.